metadata-ingestion/src/datahub/ingestion/source/salesforce.py

```diff
@@ -199,6 +212,310 @@ def report_dropped(self, ent_name: str) -> None:
 }
 
 
+class EntityDefinition(TypedDict):
+    DurableId: str
+    QualifiedApiName: str
+    DeveloperName: str
+    Label: str
+    PluralLabel: str
+    InternalSharingModel: str
+    ExternalSharingModel: str
+    DeploymentStatus: Literal[
+        "Deployed", "InDevelopment"
+    ]  # Common values for DeploymentStatus
+
+
+class UserInfo(TypedDict):
+    Username: str
+
+
+class FieldDefinition(TypedDict):
+    DataType: str
+    LastModifiedDate: str
+    LastModifiedBy: UserInfo
+    IsIndexed: bool
+    ComplianceGroup: Optional[str]
+    Description: Optional[str]
+
+
+class ReferenceTo(TypedDict):
+    referenceTo: List[str]
+
+
+class EntityParticle(TypedDict):
+    QualifiedApiName: str
+    DeveloperName: str
+    Label: str
+    DataType: str
+    Precision: Optional[int]
+    Scale: Optional[int]
+    Length: Optional[int]
+    Digits: Optional[int]
+    IsUnique: bool
+    IsCompound: bool
+    IsComponent: bool
+    ReferenceTo: Optional[ReferenceTo]
+    RelationshipName: Optional[str]
+    IsNillable: bool
+    InlineHelpText: Optional[str]
+    IsCalculated: bool
+    FieldDefinition: FieldDefinition
+
+
+class CustomObject(TypedDict):
```

Comment:
does it make sense to make these pydantic models, so that we can also validate that the API returns data in the shape we expect?

we'd likely want to use PermissiveConfigModel or BaseModel so that extra fields are ignored

Reply:
It would make sense but I'm not very confident about tracking each field's type decisively which might introduce unintended failures in currently successful flows. Current change (usage of TypedDict) only improved readability while keeping the logic exactly same.